deps/src/main/java/com/microsoft/azure/sdk/iot/deps/serializer/ErrorCodeDescription.java

```diff
@@ -0,0 +1,193 @@
+package com.microsoft.azure.sdk.iot.deps.serializer;
+
+/**
+ * See https://docs.microsoft.com/en-us/rest/api/iothub/common-error-codes for additional details
+ */
+public enum ErrorCodeDescription
+{
+    // These descriptions belong to fully qualified status codes, such as 404001
+    UnclassifiedErrorCode,
+    InvalidProtocolVersion,
+    DeviceInvalidResultCount,
+    InvalidOperation,
+    ArgumentInvalid,
+    ArgumentNull,
+    IotHubFormatError,
+    DeviceStorageEntitySerializationError,
+    BlobContainerValidationError,
+    ImportWarningExistsError,
+    InvalidSchemaVersion,
+    DeviceDefinedMultipleTimes,
+    DeserializationError,
+    BulkRegistryOperationFailure,
+    CannotRegisterModuleToModule,
+    IotHubNotFound,
+    IotHubUnauthorizedAccess,
+    IotHubUnauthorized,
+    IotHubSuspended,
+    IotHubQuotaExceeded,
+    JobQuotaExceeded,
+    DeviceMaximumQueueDepthExceeded,
+    IotHubMaxCbsTokenExceeded,
+    DeviceNotFound,
+    JobNotFound,
+    PartitionNotFound,
+    ModuleNotFound,
+    DeviceAlreadyExists,
+    ModuleAlreadyExistsOnDevice,
+    DeviceMessageLockLost,
+    MessageTooLarge,
+    TooManyDevices,
+    TooManyModulesOnDevice,
+    ThrottleBacklogLimitExceeded,
+    InvalidThrottleParameter,
+    ServerError,
+    JobCancelled,
+    ConnectionForcefullyClosedOnNewConnection,
+    DeviceNotOnline,
+    DeviceConnectionClosedRemotely,
+
+    // These descriptions belong to 3 digit http response codes such as 404.
+    BadFormat,
+    Unauthorized,
+    Forbidden,
+    NotFound,
+    Conflict,
+
+    /**
+     * Represents status code for 429 and for status code 429001
+     */
+    PreconditionFailed,
+
+    RequestEntityTooLarge,
+
+    /**
+     * Represents status code for 429 and 429001
+     */
+    ThrottlingException,
+    InternalServerError,
+
+    /**
+     * Defined for status code 503 and 503001
+     */
+    ServiceUnavailable;
+
+
+    /**
+     * Get the ErrorCodeDescription tied to the provided errorCode
+     * @param errorCode the service error code, such as 404, or 429001
+     * @return the corresponding ErrorCodeDescription
+     */
+    public static ErrorCodeDescription Parse(int errorCode)
+    {
+        switch (errorCode)
+        {
+            case(400001):
```

Comment:
Add a link to the doc that has these mappings. Is it on some MSDN doc?

Reply:
Unfortunately, I couldn't find a MSDN doc for this. These codes are taken directly from the C# SDK [here](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/iothub/service/src/Common/Exceptions/ErrorCode.cs)

Reply:
In our work with them over the track 2 SDK, we can push for a document like that, though